packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -67,18 +54,24 @@ type ContextMenuTriggerPrimitive = Polymorphic.ForwardRefComponent<
 const ContextMenuTrigger = React.forwardRef((props, forwardedRef) => {
   const { as = TRIGGER_DEFAULT_TAG, ...triggerProps } = props;
   const context = useContextMenuContext(TRIGGER_NAME);
+  const pointRef = React.useRef<Point>({ x: 0, y: 0 });
+  const virtualRef = React.useRef({
+    getBoundingClientRect: () => DOMRect.fromRect({ width: 0, height: 0, ...pointRef.current }),
+  });
+
   return (
-    <Primitive
-      {...triggerProps}
-      as={as}
-      ref={forwardedRef}
-      onContextMenu={composeEventHandlers(props.onContextMenu, (event) => {
-        event.preventDefault();
-        const point = { x: event.clientX, y: event.clientY };
-        context.onOpenChange(true);
-        context.anchorPointRef.current = point;
-      })}
-    />
+    <MenuPrimitive.Anchor virtualRef={virtualRef}>
+      <Primitive
```

Comment:
Yeah so this is the other side of the comment I left earlier. These don't really need to wrap each other right? They could be siblings I think.